beacon_node/src/config.rs

```diff
@@ -45,13 +44,6 @@ pub fn get_config<E: EthSpec>(
                 .ok_or("Failed to get freezer db path")?,
         )
         .map_err(|err| format!("Failed to remove chain_db: {}", err))?;
-
-        // Remove the pubkey cache file if it exists
-        let pubkey_cache_file = client_config.data_dir.join(PUBKEY_CACHE_FILENAME);
```

Comment:
Just to confirm, if someone runs `lighthouse bn --purge-db` the first time they use schema v3, then they'll get a v3 DB and never run the v2 migration code and leave the pubkey cache file dangling.

This seems slightly undesired, but harmless. Not worth extra code IMO.

Reply:
Yeah that's exactly right

Reply:
Perhaps worth noting that even if they revert to an old version of Lighthouse, it might try to read the undeleted pubkey cache but it will fail on the schema check